source/DistanceMap.cpp

```diff
@@ -110,27 +108,26 @@ set<const System *> DistanceMap::Systems() const
 
 
 
-// Return the destination system - the 'center' system.
-const System *DistanceMap::End() const
+// Get the planned route from center to this system.
+vector<const System *> DistanceMap::Plan(const System &system) const
 {
-	return center;
-}
-
+	auto plan = vector<const System *> {};
+	if(!HasRoute(system))
+		return plan;
 
-
-int DistanceMap::RequiredFuel(const System *system1, const System *system2) const
-{
-	auto it1 = route.find(system1);
-	auto it2 = route.find(system2);
-	if(it1 == route.end() || it2 == route.end())
-		return -1;
-	return abs(it1->second.fuel - it2->second.fuel);
+	const System *nextStep = &system;
+	while(nextStep != center)
+	{
+		plan.push_back(nextStep);
+		nextStep = route.at(nextStep).prev;
```

Comment:
Pretty sure `std::map<>.at()` throws an exception if the key you pass isn't in the map.
Is it certain that the `route` map contains this key?

Reply:
```	if(!HasRoute(system))
		return plan;```

That check means the system is in the route dictionary, meaning it is reachable from the center with a chain of RouteEdges that must exist because otherwise the system would not have been reached.